### DISCUSSION TITLE
should we add a onChanged event to the Charting Legends Component

### DISCUSSION THREAD

**Discussion by NewFuture** [microsoft/fluentui#29651] [Ideas]
I'm trying to use the [Legends](https://fluentuipr.z22.web.core.windows.net/heads/master/react-charting/demo/index.html#/examples/Legends) with third-party components.
and looking for the example to handle the selection state.

I found some similar codes with the logic to find the selectedItems in our Chart Components, like this
https://github.com/microsoft/fluentui/blob/6384a9370b8c64501f3b376a0fb177bda84bcc16/packages/react-charting/src/components/LineChart/LineChart.base.tsx#L1255

Why not expose an `onChangedEvent` in the `<Legends />` component to tell the selected items, as we already do this in the `onClick` ?
https://github.com/microsoft/fluentui/blob/6384a9370b8c64501f3b376a0fb177bda84bcc16/packages/react-charting/src/components/Legends/Legends.base.tsx#L195

**Comment by AtishayMsft** [comment]
@NewFuture Could you describe more about how you plan to use the onChangedEvent for legends.
Currently we expose the onClick event for each Legend and tie them with each chart type. 

Agree that we could make it more modular by introducing few events like onChanged, onDeselect, onMultiSelect etc.

**Reply by NewFuture** [reply]
@AtishayMsft thanks for your reply

For my case, I'm just want to know which legends are selected now.
I will be very useful for `onChange` triggering with the newest selected legends.
we do have the `onClick` event for every legend item. it's ok for single selection, but it complex for consumer to handle with multi-selected case.

For more common case, it may work like a special `Dropdown` https://developer.microsoft.com/en-us/fluentui#/controls/web/dropdown

* `onChange`: Callback issued when the selected option changes.
* `defaultSelectedKeys`: Keys that will be initially used to set selected items.
* `selectedKeys`: Keys of the selected items. If you provide this, you must maintain selection state by observing onChange events and passing a new value in when changed. Passing null in will clear the selection. Mutually exclusive with defaultSelectedKeys.

**Reply by AtishayMsft** [reply]
Sounds good.
Are you willing to contribute this functionality in the charting library.
We welcome contributions and would be happy to guide you to develop and merge the functionality to the legends component.

**Reply by NewFuture** [reply]
Sure, I will have time to see the details in the week after next.

**Comment by NewFuture** [comment]
@AtishayMsft I draft a PR implemented the `onChange` and `defaultSelectedLegends` https://github.com/microsoft/fluentui/pull/29799/files
